src/Compilers/CSharp/Portable/Binder/SubsumptionDiagnosticBuilder.cs

```diff
@@ -129,6 +129,8 @@ private ErrorCode CheckSubsumed(BoundPattern pattern, DecisionTree decisionTree,
                         // the input can no longer be null. In that case a null pattern is subsumed.
                         if (isNull && !inputCouldBeNull)
                         {
+                            // Note: we do not have any test covering this. Is it reachable?
```

Comment:
Why do we have this ```if``` then? I think we should either remove it, perhaps adding an assert, or throw from it.